---
title: "Class12 - Structural Bioinformatics"
author: "Michael Overton"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## GitHub Documents

This is an R Markdown format used for publishing markdown documents to GitHub. When you click the **Knit** button all R code chunks are run and a markdown file (.md) suitable for publishing to GitHub is generated.

## Including Code

You can include R code in the document as follows:

```{r echo=F}
PDB_stats <- read.csv("PDB_summary.csv")
PDB_stats

f_Xray <- PDB_stats[1,"Proteins"]/sum(PDB_stats$Proteins)
f_EM <- PDB_stats[3,"Proteins"]/sum(PDB_stats$Proteins)
f_Prot <- sum(PDB_stats$Proteins)/sum(PDB_stats$Total)
```

> Q: What percentage of structures in the PDB are solved by X-Ray and Electron Microscopy?
  > X-ray=89.9%
  > EM=2.1%

> Q: What proportion of structures in the PDB are protein?
  > 92.7%

> Q: Type HIV in the PDB website search box on the home page and determine how many HIV-1 protease structures are in the current PDB?
  > 141
  
  
```{r echo=F}
library(bio3d)
pdb <- read.pdb("1hsg")

pdb
attributes(pdb)
head(pdb$atom)


library(devtools)
library(BiocManager)

BiocManager::install("msa")
devtools::install_bitbucket("Grantlab/bio3d-view")

# Fetch amino acid sequence of ref Adenylate Kinase
aa <- get.seq("1ake_A")
# Blast sequence
b <- blast.pdb(aa)
# Plot hits and show breakpoint (N=16)
hits <- plot(b)
head(hits$pdb.id)

```

  
  
```{r echo=FALSE}
# Fetch files for each of the 16 HiQ matches
files <- get.pdb(hits$pdb.id, path="pdbs", split=TRUE, gzip=TRUE)
# Align releated PDBs
pdbs <- pdbaln(files, fit = TRUE, exefile="msa")

# Vector containing PDB codes for figure axis
ids <- basename.pdb(pdbs$id)

# Draw schematic alignment
plot(pdbs, labels=ids)


```
  


```{r}
library(bio3d.view)
library(rgl)

view(pdbs)
```


```{r}
anno <- pdb.annotate(ids)
unique(anno$source)

pc.xray <- pca(pdbs)
plot(pc.xray)
```


```{r}
# Calculate RMSD
rd <- rmsd(pdbs)

# Structure-based clustering
hc.rd <- hclust(dist(rd))
grps.rd <- cutree(hc.rd, k=3)

plot(pc.xray, 1:2, col="grey50", bg=grps.rd, pch=21, cex=1)
```



```{r}
#Plotting results with ggplot2
library(ggplot2)
library(ggrepel)

df <- data.frame(x=pc.xray$z[,1], y=pc.xray$z[,2])
col <- as.factor(grps.rd)

an <- anno$source

p <- ggplot(df, aes(x, y)) +
 geom_point(aes(col=col), size=2) +
 xlab("PC1") +
 ylab("PC2") +
 scale_color_discrete(name="Clusters") +
 geom_text_repel(aes(label=paste(an,"\n",ids)))
p
```



```{r}
# NMA of all structures
modes <- nma(pdbs)

plot(modes, pdbs, col=grps.rd)
```


> Q. What do you note about this plot? Are the black and colored lines similar or different? Where do you think they differ most and why?
  > The three line colors correspond to the three clusters from the PCA plot, specifically where the individual samples are located on PC1. The black lines are similar to each other and the colored lines are similar to each other. The black and colored lines differ most where there are larger structural differences.

















