---
title: "Class12 - Structural Bioinformatics"
author: "Michael Overton"
output: github_document
always_allow_html: TRUE
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Look at some statistics from the PDB database

```{r}
PDB_stats <- read.csv("PDB_summary.csv")
PDB_stats

f_Xray <- PDB_stats[1,"Proteins"]/sum(PDB_stats$Proteins)
f_EM <- PDB_stats[3,"Proteins"]/sum(PDB_stats$Proteins)
f_Prot <- sum(PDB_stats$Proteins)/sum(PDB_stats$Total)
```

Q: What percentage of structures in the PDB are solved by X-Ray and Electron Microscopy?
  > X-ray=89.9%
  > EM=2.1%

Q: What proportion of structures in the PDB are protein?
  > 92.7%

Q: Type HIV in the PDB website search box on the home page and determine how many HIV-1 protease structures are in the current PDB?
  > 141
  
Load in the protein sequence of a bacterial adenylate kinase and blast for homologs
```{r}
library(bio3d)
library(devtools)
library(BiocManager)

pdb <- read.pdb("1hsg")
pdb

aa <- get.seq("1ake_A")

```

Plot blast hits and find breakpoint
```{r}
b <- blast.pdb(aa)
hits <- plot(b)

```

  
Load in files of 16 adenylate kinases most closly related to reference and align
```{r}
match_files <- get.pdb(hits$pdb.id, path="pdbs", split=T, gzip=T, verbose=F, overwrite=T)

match_pdbs <- pdbaln(match_files, fit = T, exefile="msa")

```

Plot alignment and phylogenetic tree
```{r}
ids <- basename.pdb(match_pdbs$id)
plot(match_pdbs, labels=ids)

```

PCA plot of xray structures
```{r}
library(bio3d.view)
library(rgl)

anno <- pdb.annotate(ids)
unique(anno$source)

pc.xray <- pca(match_pdbs)
plot(pc.xray)

```

PCA plot of structures using root mean squared distance
```{r}
# Calculate RMSD
rd <- rmsd(match_pdbs)

# Structure-based clustering
hc.rd <- hclust(dist(rd))
grps.rd <- cutree(hc.rd, k=3)

plot(pc.xray, 1:2, col="grey50", bg=grps.rd, pch=21, cex=1)
```


A better plot using the ggplot package
```{r}
#Plotting results with ggplot2
library(ggplot2)
library(ggrepel)

df <- data.frame(x=pc.xray$z[,1], y=pc.xray$z[,2])
col <- as.factor(grps.rd)

an <- anno$source

p <- ggplot(df, aes(x, y)) +
 geom_point(aes(col=col), size=2) +
 xlab("PC1") +
 ylab("PC2") +
 scale_color_discrete(name="Clusters") +
 geom_text_repel(aes(label=paste(an,"\n",ids)))
p
```


```{r}
# Visualize first principal component
pc1 <- mktrj(pc.xray, pc=1, file="pc_1.pdb")

video_files <- list.files("Adenylate_kinase_pc1.mpg",
                         pattern = "\\.mpg$",
                         recursive = TRUE,
                         all.files = FALSE,
                         full.names = TRUE)

```

![](Adenylate_kinase_pc1.mpg)



Normal mode analysis of protein strucure to show regions of greatest positional change 
```{r}
modes <- nma(match_pdbs)
plot(modes, match_pdbs, col=grps.rd)

```


Q. What do you note about this plot? Are the black and colored lines similar or different? Where do you think they differ most and why?
  > The three line colors correspond to the three clusters from the PCA plot, specifically where the individual samples are located on PC1. The black lines are similar to each other and the colored lines are similar to each other. The black and colored lines differ most where there are larger structural differences.


